gui/gui/src/TGSpeedo.cxx

```diff
@@ -224,6 +231,14 @@ TGDimension TGSpeedo::GetDefaultSize() const
    return TGDimension(100, 100);
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Compute and return the mean of the circular buffer content.
+
+Float_t TGSpeedo::GetMean()
+{
+   return (Float_t) std::accumulate(fBuffer.begin(), fBuffer.end(), 0.0) / fBufferSize;
```

Comment:
/ fBuffer.size()

Reply:
It **should** be the same!

Reply:
(It's not the same if we use "reserve", and then push_back for the first N entries).
For example, if we only have filled one entry in the speedometer, fBuffer.size() would be 1, whereas fBufferSize is 10.
Otherwise, one would be doing an average of 10 values, where only 1 value has a meaning, and the other 9 are zeros from the default constructor.